pkg/util/atomicfs/fsync.go

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package atomicfs
+
+import (
+	"os"
+	"path"
+
+	"github.com/grafana/dskit/multierror"
+)
+
+// CreateFile creates a file in the filePath, write the data into the file and then execute
+// fsync operation to make sure the file and its content are stored atomically.
+func CreateFile(filePath, data string) error {
```

Comment:
This should take a `[]byte`, not `string`. This will also avoid the allocation of a string on the caller side.

Reply:
I suggested making this an `io.Reader` instead. It can be backed by whichever is more appropriate (`[]byte` or `string`). WDYT?

Reply:
That could also work, although both of the current usages will be more complex.